jobs/cjd_sample/src/main/java/com/google/samples/CompanyAndJobCrudSample.java

```diff
@@ -0,0 +1,234 @@
+            .setCompanyName(companyName)
+            .setApplicationUrls(Arrays.asList("http://careers.google.com"))
+            .setDescription("Design, develop, test, deploy, maintain and improve software.");
+    System.out.println("Job generated: " + job);
+    return job;
+  }
+  // [END basic_job]
+
+  // [START custom_attribute_job]
+
+  /**
+   * Generate a job with a custom attribute.
+   */
+  public static Job generateJobWithACustomAttribute(String companyName) throws IOException {
+    // requisition id should be a unique Id in your system.
+    String requisitionId = "jobWithACustomAttribute:" + String.valueOf(new Random().nextLong());
+
+    // Constructs custom attributes map
+    Map<String, CustomAttribute> customAttributes = new HashMap<>();
+    customAttributes.put(
+        "some_field_name",
+        new CustomAttribute()
+            .setStringValues(new StringValues().setValues(Arrays.asList("value1")))
+            .setFilterable(true));
+
+    // Creates job with custom attributes
+    Job job =
+        new Job()
+            .setCompanyName(companyName)
+            .setRequisitionId(requisitionId)
+            .setJobTitle("Software Engineer")
+            .setApplicationUrls(Arrays.asList("http://careers.google.com"))
+            .setDescription("Design, develop, test, deploy, maintain and improve software.")
+            .setCustomAttributes(customAttributes);
+    System.out.println("Job generated: " + job);
+    return job;
+  }
+  // [END custom_attribute_job]
+
+  // [START create_job]
+
+  /**
+   * Create a job.
+   */
+  public static Job createJob(Job jobToBeCreated) throws IOException {
+    CreateJobRequest createJobRequest = new CreateJobRequest().setJob(jobToBeCreated);
+    Job jobCreated = jobService.jobs().create(createJobRequest).execute();
+    System.out.println("Job created: " + jobCreated);
+    return jobCreated;
+  }
+  // [END create_job]
+
+  // [START get_job]
+
+  /**
+   * Get a job.
+   */
+  public static Job getJob(String jobName) throws IOException {
+    Job jobExisted = jobService.jobs().get(jobName).execute();
+    System.out.println("Job existed: " + jobExisted);
+    return jobExisted;
+  }
+  // [END get_job]
+
+  // [START update_job]
+
+  /**
+   * Update a job.
+   */
+  public static Job updateJob(String jobName, Job jobToBeUpdated) throws IOException {
+    UpdateJobRequest updateJobRequest = new UpdateJobRequest().setJob(jobToBeUpdated);
+    Job jobUpdated = jobService.jobs().patch(jobName, updateJobRequest).execute();
+    System.out.println("Job updated: " + jobUpdated);
+    return jobUpdated;
+  }
+  // [END update_job]
+
+  // [START delete_job]
+
+  /**
+   * Delete a job.
+   */
+  public static void deleteJob(String jobName) throws IOException {
+    jobService.jobs().delete(jobName).execute();
+    System.out.println("Job deleted");
+  }
+  // [END delete_job]
+
+  public static void main(String... args) throws Exception {
```

Comment:
The pattern in main examples is to have some kind of argument parsing that allows the user to run the sample with some command line arguments indicative of what they want to do.

So the first arg would be one of createJob, updateJob, deleteJob, createCompany, updateCompany, deleteCompany... and then the sample would use the Options class to parse the options relevant to that sample (i.e. requiring company name, etc when creating a company).

Having a main method that basically creates, updates, and deletes a company and a job is kind of a redundant test method that just spits out output to the console for the user that isn't ultimately that instructional.

Reply:
Our concern is that CRUD is related. If we let the customer input the parameter themselves they might configure them incorrectly. Besides, job creation is related, they need to create the company before creating the job. For our new customer, they might not know this point.